### DISCUSSION TITLE
VHD ( camera device) update using genric UVC updater

### DISCUSSION THREAD

**Discussion by sureshmanisns** [fwupd/fwupd#5293] [General]
**Describe the question**
Is there a any generic plugin which can be used to update video/camera device using fuwpd?

**fwupd version information**
fwupd 1.7.9 and above version
```

**Comment by hughsie** [comment]
There are a lot of different update protocols used by each vendor. The closest thing to a generic update protocol for peripherals is DFU but most vendors don't use that. So no, no generic plugin I'm afraid.

**Reply by sureshmanisns** [reply]
> There are a lot of different update protocols used by each vendor. The closest thing to a generic update protocol for peripherals is DFU but most vendors don't use that. So no, no generic plugin I'm afraid.

Ok thank you. DFU plugin enumerates my Camera device as well.

**Reply by hughsie** [reply]
Can you share the make/model? The vendor might already have an LVFS account.

**Reply by sureshmanisns** [reply]
> Can you share the make/model? The vendor might already have an LVFS account.

epos expand vision 1

**Reply by hughsie** [reply]
EPOS has uploaded quite a lot of firmware to the LVFS, but not that device I'm afraid.

**Reply by sureshmanisns** [reply]
> EPOS has uploaded quite a lot of firmware to the LVFS, but not that device I'm afraid.

ok got it. Is there a any plug-in in fwupd, which is related to VHD/ValueHD/UVC?

**Comment by hughsie** [comment]
There's the logitech-scribe and synaptics-cape plugins which might be what you're looking for.
